src/main/java/com/exadel/discountwebapp/user_discount/service/UserDiscountService.java

```diff
@@ -0,0 +1,43 @@
+package com.exadel.discountwebapp.user_discount.service;
+
+import com.exadel.discountwebapp.exception.exception.client.EntityAlreadyExistsException;
+import com.exadel.discountwebapp.exception.exception.client.EntityNotFoundException;
+import com.exadel.discountwebapp.user_discount.entity.UserDiscount;
+import com.exadel.discountwebapp.user_discount.mapper.UserDiscountMapper;
+import com.exadel.discountwebapp.user_discount.repository.UserDiscountRepository;
+import com.exadel.discountwebapp.user_discount.vo.UserDiscountRequestVO;
+import com.exadel.discountwebapp.user_discount.vo.UserDiscountResponseVO;
+import com.google.zxing.WriterException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+
+@Service
+@RequiredArgsConstructor
+public class UserDiscountService {
+
+    private final UserDiscountRepository userDiscountRepository;
+    private final UserDiscountMapper userDiscountMapper;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public byte[] create(UserDiscountRequestVO request) throws IOException, WriterException {
+        UserDiscount userDiscount = userDiscountMapper.toEntity(request);
+        if (userDiscountRepository.existsById(new UserDiscount.UserDiscountId(request.getUserId(), request.getDiscountId()))) {
```

Comment:
this check should be before toEntity method

Reply:
Done